### PR TITLE
Imread: use `da.map_blocks` instead of `da.concatenate`

### PR DESCRIPTION
**Pull request by m-albert**
This PR addresses performance improvements for array creation discussed in https://github.com/dask/dask-image/issues/161 and leads to arrays with smaller dask graphs (see https://github.com/dask/dask/issues/5913).

### CODE REVIEW

**Review by jni — APPROVED** [review]

File: dask_image/imread/__init__.py [hunk 1]
```diff
@@ -71,21 +71,20 @@ def imread(fname, nframes=1, *, arraytype="numpy"):
...
+        arrayfunc=arrayfunc,
+        meta=arrayfunc([]).astype(dtype),  # meta overwrites `dtype` argument
+    )
 
     return a
+
+
+def _map_read_frame(block_info=None, **kwargs):
+
+    i, j = block_info[None]['array-location'][0]
```
Comment (jakirkham) [inline dask_image/imread/__init__.py:88]: Why is indexing by `None` needed?

Comment (jakirkham) [reply dask_image/imread/__init__.py:88]: NVM now see this in the docs. This is just for the expected output array's info.

**Review by jakirkham — APPROVED** [review]

**Comment by GenevieveBuckley** [comment]
Thanks @m-albert 🚀

**Comment by jni** [comment]
❤️ 🚀 🚀 🚀

**Comment by jakirkham** [comment]
May be worthing seeing if we can build off this work ( https://github.com/dask/dask/pull/6931 ).

**Pull request merged**